src/nested_artboard.cpp

```diff
@@ -110,3 +113,31 @@ void NestedArtboard::update(ComponentDirt value) {
         m_Artboard->opacity(renderOpacity());
     }
 }
+
+bool NestedArtboard::hasNestedStateMachines() const {
+    for (auto animation : m_NestedAnimations) {
+        if (animation->is<NestedStateMachine>()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+const std::vector<NestedAnimation*>& NestedArtboard::nestedAnimations() const {
+    return m_NestedAnimations;
+}
+
+bool NestedArtboard::worldToLocal(Vec2D world, Vec2D* local) {
+    assert(local != nullptr);
+    if (m_Artboard == nullptr) {
```

Comment:
interesting. how could this be?

Reply:
Oh right, also the artboard could be null if someone created a nested artboard but never selected an artboard to nest into it. We can't just drop these at export because they might also be using them as a target for a constraint, grouping, draw order, clip, or some other Transform node-like behavior else (since they're Nodes).